### DISCUSSION TITLE
Edge and NextDNS

### DISCUSSION THREAD

**Discussion by ghost** [HotCakeX/Harden-Windows-Security#105] [Q&A]
Hello! 
Love the Project! just a quick question. If I use a NextDNS profile for Ad blocking and parental controls, how would I change the module to use NextDNS instead of the initial cloudflare?

**Comment by HotCakeX** [comment]
Hi,
Thank you!

The Harden Windows Security module in this repository doesn't touch DNS. I think what you are referring to is this module in another repo:
https://github.com/HotCakeX/WinSecureDNSMgr

is that correct?

**Reply by ghost** [reply]
My mistake! I thought that in the script it sets Edges parameters to be ultra secure including changing the DNS  ( DOH, etc)

**Reply by HotCakeX** [reply]
Oh no worries, if you apply the hardening measures, Edge will use system DNS. So if you set NextDNS in Windows settings, Edge will use that, along with all other applications on your computer. This is a more secure method. This way the Application Guard in Edge will also use NextDNS:  https://github.com/HotCakeX/Harden-Windows-Security?tab=readme-ov-file#edge-browser-configurations

**Reply by ghost** [reply]
Great! So after I set the script I just need to go in the DNS settings and manually change it my NextDNS profile?

**Reply by HotCakeX** [reply]
Yes ^^
Either before or after, they won't be changed by the script. I use NextDNS too btw
Made a PowerShell script that lets me view the NextDNS logs in real time on PowerShell console, including which domain was blocked etc.
https://github.com/HotCakeX/MicrosoftDomains/blob/main/NextDNS%20API/Stream%20the%20logs%20-%20Customized%20Output.ps1

**Reply by ghost** [reply]
Oh that is awesome!

**Reply by ghost** [reply]
I do have a question about NextDNS. Currently if you set the parental controls to “block bypass methods” you can still download the VPN app ( mullvad, IVPN, etc) it will bypass that feature. Do you have a way for the apps to also be blocked? I feel it defeats the purpose of that actual feature. I also raised this issue with NextDNS.

**Reply by HotCakeX** [reply]
This is usually a cat and mouse game, they don't usually block every possible way to use VPNs, even governments can't do it, let alone NextDNS service.

Parental control is very easy on Windows. If you use WDAC you will be able to only allow certain apps on the device and nothing else will run, but on phones I'm not too sure.

**Reply by ghost** [reply]
Oh I see. Well that is disappointing. Thank you so much for your help though!

**Reply by HotCakeX** [reply]
You're welcome! :)
